rpcs3/Emu/system_progress.hpp

```diff
@@ -59,11 +60,12 @@ class scoped_progress_dialog final
 	scoped_progress_dialog& operator=(const std::string& text) noexcept
 	{
 		ensure(!text.empty());
-		m_text = text;
 
 		// This method is destroying the previous value and replacing it with a new one
-		std::tie(m_prev, m_prev_id, m_id) = g_progr_text.atomic_op([this](progress_dialog_string_t& progr)
+		std::tie(m_prev, m_prev_id, m_id) = g_progr_text.atomic_op([this, text](progress_dialog_string_t& progr)
 		{
+			m_text = text;
```

Comment:
1) You imply that operator= is called concurrently on the same object? 
2) If 1 is true, then how it fixes it?

Reply:
I think I saw instances where it's updated inside worker threads

Reply:
Where? this should be a local object, it's not meant for this